ts/util/table-builder.ts

```diff
@@ -374,6 +381,23 @@ export class TableBuilder<ParentType, ChildType> {
     };
   }
 
+  private addLegendControls(container: HTMLElement) {
+    const legendButton = document.createElement("button");
+    legendButton.className =
+      "bg-green-200 rounded-md hover:ring-2 ring-offset-1 ring-green-200 text-white font-inter font-medium text-12 px-2 py-1";
+    legendButton.innerHTML = "Legend";
+    legendButton.onclick = () => {
+      const legendWindow = document.getElementById("legend-container");
+      if (!legendWindow) {
+        const legendContainer = new Legend();
+        this.container.appendChild(legendContainer.getTag());
+      } else {
+        legendWindow.remove();
+      }
+    };
+    container.appendChild(legendButton);
```

Comment:
Instead of having this code here, could there be a `toggleLegend` function in `legend.ts`? Don't even need to pass it the container since you can append it to the document body instead and it shouldn't make any difference